lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -1249,27 +1249,9 @@ def determine_group_level_monitored_resource_and_labels(tag,
       # Examples:
       # "container.<container_id>" // Docker container.
       # "k8s_pod.<namespace_name>.<pod_name>" // GKE pod.
-      if @enable_metadata_agent && local_resource_id
-        @log.debug 'Calling metadata agent with local_resource_id: ' \
-                  "#{local_resource_id}."
-        retrieved_resource = query_metadata_agent_for_monitored_resource(
-          local_resource_id)
-        @log.debug 'Retrieved monitored resource from metadata agent: ' \
-                  "#{retrieved_resource.inspect}."
-        if retrieved_resource
-          resource = retrieved_resource
-          # TODO(qingling128): Fix this temporary renaming from 'gke_container'
-          # to 'container'.
-          resource.type = 'container' if resource.type == 'gke_container'
-        else
-          # TODO(qingling128): This entire else clause is temporary before we
-          # implement buffering and caching.
-          @log.warn('Failed to retrieve monitored resource from Metadata' \
-                     " Agent with local_resource_id #{local_resource_id}.")
-          constructed_k8s_resource = construct_k8s_resource_locally(
-            local_resource_id)
-          resource = constructed_k8s_resource if constructed_k8s_resource
-        end
+      if local_resource_id
+        global_resource = convert_local_resource_id_globally(local_resource_id)
```

Comment:
I personally feel that "convert_local_resource_id_to_monitored_resource" is clearer, but I'd love to hear why you chose "globally"

Reply:
`convert_local_resource_id_to_monitored_resource` sounds better. Changed.

Reply:
[Optional] This could even have been `monitored_resource_from_local_id` (somewhat shorter).

Reply:
Missed this one. Renamed.